custom_components/husqvarna_automower_ble/__init__.py

```diff
@@ -28,16 +28,19 @@
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up Husqvarna Autoconnect Bluetooth from a config entry."""
     address = entry.data[CONF_ADDRESS]
-    #pin = entry.data[CONF_PIN]
+    pin = entry.data[CONF_PIN]
     channel_id = entry.data[CONF_CLIENT_ID]
 
     LOGGER.info(STARTUP_MESSAGE)
 
-    mower = Mower(channel_id, address)
+    if pin != 0:
+        mower = Mower(channel_id, address, pin)
+    else:
+        mower = Mower(channel_id, address)
 
     await close_stale_connections_by_address(address)
 
-    LOGGER.debug("connecting to %s with channel ID %s", address, str(channel_id))
+    LOGGER.debug("connecting to %s with channel ID %s", address, str(channel_id), str(pin))
```

Comment:
```suggestion
    LOGGER.debug("connecting to %s with channel ID %s, PIN: %s", address, str(channel_id), str(pin))
```

Reply:
Good spot thanks @olekenneth 
I've actually redone this code anyway in the dev7 branch so I'm going to close this pr to avoid confusion.

The Dev7 is working for me at home right now so once I've done more testing I'll push that to live.